### PR TITLE
feat: [CDS-47127]: force delete template

### PR DESCRIPTION
**Pull request by RajBaviskar**
### Summary

* Adding modal for template force delete 

#### Screenshots

Screen Shot 2023-01-05 at 3 22 44 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by RajBaviskar** [comment]
retrigger titlecheck

**Comment by RajBaviskar** [comment]
retrigger jest

**Comment by RajBaviskar** [comment]
retrigger titlecheck

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx [hunk 1]
```diff
@@ -127,16 +136,45 @@ export const DeleteTemplateModal = (props: DeleteTemplateProps) => {
...
+        }
+      } else {
+        showError(
+          getRBACErrorMessage(err as RBACError),
+          undefined,
+          areMultipleVersionsSelected
+            ? 'common.template.deleteTemplate.errorWhileDeletingTemplates'
+            : 'common.template.deleteTemplate.errorWhileDeletingTemplate'
+        )
+      }
```
Comment (rameshx) [inline src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:159]:
```suggestion
      if (
        CDS_FORCE_DELETE_ENTITIES &&
        err?.data?.code === "ENTITY_REFERENCE_EXCEPTION"
      ) {
        openReferenceErrorDialog()
        return
      }
      
      showError(
        getRBACErrorMessage(err as RBACError),
        undefined,
        areMultipleVersionsSelected
          ? "common.template.deleteTemplate.errorWhileDeletingTemplates"
          : "common.template.deleteTemplate.errorWhileDeletingTemplate"
      );
```

File: src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx [hunk 2]
```diff
@@ -96,7 +99,12 @@ export const DeleteTemplateModal = (props: DeleteTemplateProps) => {
     }
   }, [templatesError])
 
-  const performDelete = async (commitMsg: string, versions?: string[]): Promise<void> => {
+  const [commitMessage, setCommitMessage] = React.useState<string>(`${getString('delete')} ${template.name}`)
+  React.useEffect(() => {
+    setCommitMessage(`${getString('delete')} ${template.name}`)
+  }, [getString, template])
```
Comment (rameshx) [inline src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:105]: state and effect are not needed here

```suggestion
  const commitMessage = `${getString('delete')} ${template.name}`
```

File: src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx (hunk 1 above)
Comment (RajBaviskar) [reply src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:159]: changed.

File: src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx (hunk 2 above)
Comment (RajBaviskar) [reply src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:105]: changed.

File: src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx [hunk 3]
```diff
@@ -137,6 +147,21 @@ export const DeleteTemplateModal = (props: DeleteTemplateProps) => {
     }
   }
 
+  const redirectToReferencedBy = (): void => {
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:150]: This function isn't doing anything

Comment (AshwinBhatkal-Harness) [reply src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:150]: We can directly use closeDialog

Comment (RajBaviskar) [reply src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:150]: Its required by useEntityDeleteErrorHandlerDialog

File: src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx [hunk 4]
```diff
@@ -137,6 +147,21 @@ export const DeleteTemplateModal = (props: DeleteTemplateProps) => {
...
+    closeDialog()
+  }
+
+  const { openDialog: openReferenceErrorDialog, closeDialog } = useEntityDeleteErrorHandlerDialog({
+    entity: {
+      type: ResourceType.TEMPLATE,
+      name: defaultTo(template?.name, '')
+    },
+    redirectToReferencedBy: redirectToReferencedBy,
```
Comment (AshwinBhatkal-Harness) [inline src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:159]:
```suggestion
    redirectToReferencedBy: closeDialog,
```

Comment (RajBaviskar) [reply src/modules/72-templates-library/components/DeleteTemplateModal/DeleteTemplateModal.tsx:159]: Screen Shot 2023-01-11 at 1 48 46 AM

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

**Review by samarth-sharma31 — DISMISSED** [review]

**Review by douglasnfrazao — APPROVED** [review]

**Review by rameshx — APPROVED** [review]

**Review by abhinav-harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**